sale_coupon_multi_use/wizards/sale_coupon_apply_code.py

```diff
@@ -0,0 +1,19 @@
+from odoo import models
+
+
+class SaleCouponApplyCode(models.TransientModel):
+    """Extend to modify apply_coupon method for multi coupon usage."""
+
+    _inherit = "sale.coupon.apply.code"
+
+    def apply_coupon(self, order, coupon_code):
+        """Extend to pass order coupon ctx for multi coupon usage."""
+        self = self.with_context(
+            coupon_order_data={
```

Comment:
I don't really like using context to pass arguments. It was not done for that purpose and should be avoided as much as possible (I know the original apply_coupon is not easily overridable)

Reply:
So any suggestion, to extend it better? 
Because in my opinion otherwise you would need to overwrite copy/paste blocks of code, which is ugly and not great.
Also, for what else, context is meant, if not passing stuff around? It does just pass stuff:) Though I agree its not as obvious as if you would pass proper argument from method.

I use this rule of thumb (priority):
- Extend method normally and pass/modify it without context (like for example, you can update result dictionary or something).
- If can't reach needed insides of a method by extending normally, pass context to intercept wanted parts.
- if context can't be used (not reachable, poor original implementation etc), then use last resort override whole method.

Reply:
No, I've not digged into code too much. If you estimate this is the only way to do it, let's go. Just change the other things.